travis/travis_weblate.py

```diff
@@ -1,166 +1,216 @@
+            cmd = ("sed -i -e \"s/row\\['translation'] = src/"
+                   "row['translation'] = ''/g\" %s" % s_file)
+            print cmd
+            subprocess.call(cmd, shell=True)
+
+    def _generate_odoo_po_files(self):
+        with self._connection_context(self._server_path, self._addons_path,
+                                      "openerp_test") as odoo_context:
+            for module in self._addons_list:
+                print("\n", yellow("Obtaining POT file for %s" % module))
+                i18n_folder = os.path.join(self._travis_build_dir, module,
+                                           'i18n')
+                if not os.path.isdir(i18n_folder):
+                    os.makedirs(i18n_folder)
+                # Put git add for letting known git which translations to update
+                po_files = glob.glob(os.path.join(i18n_folder, '*.po'))
+                for lang in self._langs:
+                    if os.path.isfile(os.path.join(i18n_folder, lang + '.po')):
+                        continue
+                    with open(os.path.join(i18n_folder, lang + '.po'), 'wb') \
+                            as f_po:
+                        f_po.write(odoo_context.get_pot_contents(module, lang))
+                for po_file_name in po_files:
+                    lang = os.path.basename(os.path.splitext(po_file_name)[0])
+                    if self._langs and lang not in self._langs:
+                        # Limit just allowed languages if is defined
+                        continue
+                    po_file_path = os.path.join(i18n_folder, po_file_name)
+                    with open(po_file_path, 'r') as f_po:
+                        odoo_context.load_po(f_po, lang)
+                    new_content = odoo_context.get_pot_contents(module, lang)
+                    with open(po_file_path, 'wb') as f_po:
+                        f_po.write(new_content)
+
+    def _add_odoo_po_files(self, component):
+        po_return = []
+        po_files = glob.glob(os.path.join(self._travis_build_dir,
+                                          component['filemask']))
+        status = self._git.run(["status"])
+        for po_file in po_files:
+            po = po_file.replace(os.getcwd(), '')[1:]
+            if po in status:
+                po_return.append(po_file)
+        if po_return:
+            self._git.run(["add"] + po_return)
+        return True if po_return else False
+
+    def _last_modifier(self, repo):
+        login = ''
+        author_email = (self._git.run(
+            ["show", "--pretty=format:%ae", repo]).split('\n')[0])
+        if author_email:
+            user = self.gh_api.get_user_info(author_email)
+            if 'login' in user:
+                login = '@%s\n' % user['login']
+        return login
+
+    def _has_conflict(self, component):
+        status = self._git.run(["status"])
+        if 'both modified' in status:
+            self._register_pull_request(component, status)
```

Comment:
I made two method the `_check_conflict` method and `_register_pull_request` method